source/guide_grafana-loki.rst

```diff
@@ -0,0 +1,277 @@
+
+Directory for storing the custom configuration files:
+
+::
+
+ [isabell@stardust ~]$ mkdir -p ~/etc/loki
+ [isabell@stardust ~]$
+
+Directory for storing the log files:
+
+::
+
+ [isabell@stardust ~]$ mkdir -p ~/var/log/grafana
+ [isabell@stardust ~]$
+
+Loki Installation
+=================
+
+Clone loki from the official git repository:
+
+::
+
+ [isabell@stardust ~]$ git clone https://github.com/grafana/loki ~/software/
+ [isabell@stardust ~]$ cd ~/software/loki
+ [isabell@stardust loki]$
+
+Build loki
+----------
+
+::
+
+ [isabell@stardust loki]$ make loki
+
+::
+
+Move the binary to ``~/bin`` and the default configuration file to ``~/etc/loki``.
+
+::
+
+ [isabell@stardust loki-2.6.1]$ mv cmd/loki/loki ~/bin/
+ [isabell@stardust loki-2.6.1]$ mv cmd/loki/loki-local-config.yaml ~/etc/loki/loki.yaml
+
+Loki Configuration
+------------------
+
+
+Create the file ``~/etc/loki/loki.yaml`` with the following content:
+
+.. code-block:: yaml
+  :emphasize-lines: 4,5,7,13,14
+
+  auth_enabled: false
+
+  server:
+    http_listen_address: isabell.local.uberspace.de
+    http_listen_port: 3100
+    grpc_listen_port: 9096
+
+  common:
+    path_prefix: /home/isabell/tmp/loki
+    storage:
+      filesystem:
+        chunks_directory: /home/isabell/tmp/loki/chunks
+        rules_directory: /home/isabell/tmp/loki/rules
+    replication_factor: 1
+    ring:
+      instance_addr: 127.0.0.1
+      kvstore:
+        store: inmemory
+
+  schema_config:
+    configs:
+      - from: 2020-10-24
+        store: boltdb-shipper
+        object_store: filesystem
+        schema: v11
+        index:
+          prefix: index_
+          period: 24h
+
+  ruler:
+    alertmanager_url: http://localhost:9093
+
+.. note:: The ``http_listen_address`` must be configured to listen on the local network interface of your uberspace account, so that other applications like Pronmtail (see below) can access Loki via HTTP.
+
+Setup daemon
+------------
+
+Create the file ``~/etc/services.d/loki.ini`` with the following content:
+
+.. code-block:: ini
+
+  [program:loki]
+  command=loki
+    -config.file %(ENV_HOME)s/etc/loki/loki.yaml
+  autostart=yes
+  autorestart=yes
+
+What the arguments for loki mean:
+
+  * ``-config.file``: The location of the custom configuration file we created.
+
+Connecting Grafana to loki
+--------------------------
+
+.. note:: At that point, you already should have installed :lab:`Grafana <guide_grafana>`.
+
+Go to the main page of your Grafana installation and navigate to Configuration / Data sources.
+
+Click "Add data source" and select "Loki". Choose a name you like and add ``http://localhost:3100`` as HTTP URL.
+
+Promtail
+========
+
+Promtail is the logshipper in the Grafana ecosystem. It scrapes logs periodically and sends them to loki.
+
+Install Promtail
+----------------
+
+::
+
+  [isabell@stardust ~]$ mkdir -p ~/etc/promtail
+  [isabell@stardust ~]$ wget https://github.com/grafana/loki/releases/download/v2.4.2/promtail-linux-amd64.zip
```

Comment:
this is a fixed version, there should be a hint and url to check for latest version and replace the version number